Raise exception instead of producing empty screenshots

imsodin opened the pull request:
Screenshotting through `.show` produces the desired images, but calling `.screenshot` just produces empty files. That's because of the `._first_time` variable preventing actual rendering. In show it is set to `False`, thus it works -> do it here too.

GuillaumeFavelier commented:
I think you describe the scenario when one calls `screenshot()` without actually calling `show()` ? 

This seems to be on purpose. [In the documentation](https://github.com/pyvista/pyvista/blob/b2ce8318c5d36ac9032173bc8475cb1a62a27bac/pyvista/plotting/plotting.py#L745-L749), I read:

```py
    def render(self):
        """Render the main window.

        If this is called before ``show()``, nothing will happen.
        """
```

It's related to the design, what do you think @banesullivan ?

imsodin commented:
Yes I do.

The `screenshot` documentation doesn't say you need to call `show` first, neither is it done in the example: https://docs.pyvista.org/plotting/plotting.html#pyvista.BasePlotter.screenshot  
If you aren't supposed to use `screenshot` before `show`, it should raise an error, not silently produce empty output. However that would be an annoying behaviour for me: I want to create images programmatically without having to open an interactive window.

GuillaumeFavelier commented:
> I want to create images programmatically without having to open an interactive window.

Did you try with `off_screen=True`?

The following works for me and I don't need to call `show()` :

```py
import pyvista
sphere = pyvista.Sphere()
plotter = pyvista.Plotter(off_screen=True)  # same as the example but off_screen=True
actor = plotter.add_mesh(sphere)
plotter.screenshot('screenshot.png')
```

imsodin commented:
That works - thanks.

I reverted my change, updated the `screenshot` example to reflect this and added an exception telling the user about calling `.show` first or using `off_screen` instead of saving blank screenshots.

banesullivan commented:
Thanks, @GuillaumeFavelier. This was indeed a design decision we made a long while back. It's tough to meet all use cases and this implementation was one that generally works all around stably.

I think we'd need to do some serious evaluation if changing this behaviour

imsodin commented:
I am not changing the behaviour anymore, now that I understand the rationale (thanks again for explaining). This PR now only attempts to improve UX through adjusting the documentation example and raising a meaningful error instead of producing blank screenshots.

I don't understand the docs CI failure, it runs fine locally (python3.6 and vtk 8.2 on conda).

banesullivan reviewed (approved):
Looks good to me!

imsodin commented:
I don't remember the merge/PR policy: Do you squash on merge or do I need to squash my commits before merging?

akaszynski commented:
Squash and merge as it keeps the commit history for master cleaner.

imsodin commented:
Then this is done from my side -  just in case anyone is waiting for input from me before merging ;)
